src/mutable_data.rs

```diff
@@ -214,11 +220,22 @@ macro_rules! impl_mutable_data {
                 self.permissions.get(&user).ok_or(Error::NoSuchKey)
             }
 
-            fn check_permissions(&self, request: Request, requester: PublicKey) -> Result<()> {
+            fn check_permissions(
+                &self,
+                request: Request,
+                msg_id: MessageId,
+                requester: PublicKey,
+                signature: Signature,
+            ) -> Result<()> {
+                verify_signature(signature, requester, request.clone(), msg_id)?;
```

Comment:
I think this is better done outside of the `check_permissions` function.
So now it does 2 things while it can do just 1 and don't really care about the signatures.
What I'd propose is to leave `verify_signature` as it is, and remove the signature checking and the `signature` argument here.